doc/lua_api.md

```diff
@@ -7472,6 +7472,19 @@ child will follow movement and rotation of that bone.
                 abides by, `"custom"` uses `sun_tint` and `moon_tint`, while
                 `"default"` uses the classic Minetest sun and moon tinting.
                 Will use tonemaps, if set to `"default"`. (default: `"default"`)
+        * `fog`: A table with following optional fields:
+            * `fog_distance`: integer, override the client's viewing_range.
+               By default, fog_distance is controlled by the client's viewing_range, and this field is not set.
+               When set the client client can no longer set a larger viewing_range (including range_all).
+               Any value >= 0 sets the desired fog_distance.
+               Any value < 0, resets the behavior to being client-controlled.
+               (default: not set)
+            * `fog_start`: float, override the client's fog_start.
+               Fraction of the visible distance at which fog starts to be rendered.
```

Comment:
Hmm... Then we remove this feature and the client setting.

Reply:
>Can't we just do what you did here: https://github.com/Desour/minetest/tree/test_exp_fog

In case you're referring to the `fog_density = fogShadingParameter * 2.0;`, that's just something that I found by trying out, to keep the view somewhat the same. It probably won't work in general. And it limits the possible range for the fog density.

A big benefit of exponential squared fog over linear fog would be that one could make the fog much denser, btw., (i.e. `fog_start=0` with linear fog is still very clear).

Reply:
FYI, if you want something for testing exp2 fog, here's a small branch: https://github.com/Desour/minetest/tree/lhofhansl_fog_control_exp2 :-)

At this point, I'm unsure how a good API for fog density control would even look like, e.g. should the density be relative to the view distance or not?
Sorry for making this more complicated.

Reply:
With exponential fog you also have to careful that fog just lines up with the viewing range. I.e. if the viewing_range is 500 the exponential fog to be chosen so that it is opaque exactly at 500. Otherwise you either load unnecessary blocks or you can see the jagged edge of the world.

With this in mind I suggest as long as we have linear fog we use that. Same the client fog_start setting.

Reply:
Ok, let's keep the `fog_start` for now then.